### PR TITLE
ref(ts) Convert EventDataSection to Typescript

### PR DESCRIPTION
**Pull request by markstory**

### CODE REVIEW

File: src/sentry/static/sentry/app/components/events/eventDataSection.tsx [hunk 1]
```diff
@@ -3,7 +3,26 @@ import React from 'react';
...
+};
+
+type DefaultProps = Readonly<typeof defaultProps>;
+
+type Props = {
+  className?: string;
+  title: React.ReactText;
+  type: string;
+  wrapTitle: boolean;
+  toggleRaw: Function;
```
Comment (scefali) [inline src/sentry/static/sentry/app/components/events/eventDataSection.tsx]: @markstory any reason you don't use a function signature here?

File: src/sentry/static/sentry/app/components/events/eventDataSection.tsx [hunk 2]
```diff
@@ -3,7 +3,26 @@ import React from 'react';
...
+  raw: boolean;
+  hideGuide: boolean;
+} & Partial<DefaultProps>;
```
Comment (scefali) [inline src/sentry/static/sentry/app/components/events/eventDataSection.tsx]: @markstory What's the advantage of doing `& Partial<DefaultProps>` when there are already types for the props?

File: src/sentry/static/sentry/app/components/events/eventDataSection.tsx (hunk 1 above)
Comment (markstory) [reply src/sentry/static/sentry/app/components/events/eventDataSection.tsx]: I'm dumb. This doesn't need to be here it is also defined in DefaultProps.

File: src/sentry/static/sentry/app/components/events/eventDataSection.tsx (hunk 2 above)
Comment (markstory) [reply src/sentry/static/sentry/app/components/events/eventDataSection.tsx]: If I wasn't a muppet it lets you not have to redefine the types for default props. I'll clean that up.

Comment (scefali) [reply src/sentry/static/sentry/app/components/events/eventDataSection.tsx]: https://vignette.wikia.nocookie.net/disney/images/2/26/Walter.png/revision/latest?cb=20111013192451

**Review by scefali — APPROVED** [review]

**Pull request merged**